python_modules/dagster/dagster/_core/workspace/load_target.py

```diff
@@ -41,21 +41,44 @@ def create_origins(self) -> Sequence[CodeLocationOrigin]:
         return location_origins_from_yaml_paths(self.paths)
 
 
+def is_valid_modules_list(modules: Any) -> bool:
+    if isinstance(modules, list):
+        for item in modules:
+            if not (isinstance(item, dict) and 'type' in item and 'name' in item):
+                return False
+        return True
+    return False
+
```

Comment:
Similarly if there is a `modules` block and it is incorrectly typed, let's throw a hard error so that the user knows they have done something wrong,

Reply:
@schrockn I extended the code here to fail harder, as well as checking a bit more in detail for the contents. Added tests for it